os/kernel/binary_manager/binary_manager_resource.c

```diff
@@ -0,0 +1,132 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <debug.h>
+#include <stdlib.h>
+#include <queue.h>
+#include <errno.h>
+#include <mqueue.h>
+#ifdef CONFIG_FS_NAMED_SEMAPHORES
+#include <semaphore.h>
+#endif
+#include <sys/types.h>
+#include <tinyara/mm/mm.h>
+#include <tinyara/sched.h>
+#include <tinyara/init.h>
+
+#include "sched/sched.h"
+#include "binary_manager.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+struct resource_node_s {
+	struct resource_node_s *flink;
+	int type;
+	char name[CONFIG_NAME_MAX];
+};
+typedef struct resource_node_s resource_node_t;
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/ 
+/****************************************************************************
+ * Name: binary_manager_clear_bin_inode
```

Comment:
Please provide function description and input/output parameters also.

Reply:
Done